From f8d9ce30cc36e212def45f7ad259fe3a6b4cc19a Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Mon, 18 Jan 2021 08:30:11 +0100
Subject: [PATCH 32/35] configs, topic_miami[lite]: Fix boot order and disable
 BOARD_LATE_INIT

The BOARD_LATE_INIT option for the Xilinx zynq prepends the boot_targets with
the current device. For the mimai board, we want a fixed order: First try
booting with mmc0 (SD card), then try QSPI, regardless of the boot switch
setting. This is especially important for the miamilite which does not have
bootswitches.

Fix boot order, "qspi0" should be "qpsi", don't try JTAG first, and remove
"usb1" which doesn't exist.
---
 configs/topic_miami_defconfig     | 1 +
 configs/topic_miamilite_defconfig | 1 +
 include/configs/topic_miami.h     | 2 +-
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/configs/topic_miami_defconfig b/configs/topic_miami_defconfig
index c0ae3bec58..9bcbb8c952 100644
--- a/configs/topic_miami_defconfig
+++ b/configs/topic_miami_defconfig
@@ -16,6 +16,7 @@ CONFIG_DEBUG_UART_CLOCK=100000000
 CONFIG_BOOT_INIT_FILE="board/topic/zynq/zynq-topic-miami/ps7_regs.txt"
 CONFIG_DEFAULT_DEVICE_TREE="zynq-topic-miami"
 CONFIG_DEBUG_UART=y
+# CONFIG_BOARD_LATE_INIT is not set
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_SYS_CUSTOM_LDSCRIPT=y
 CONFIG_SYS_LDSCRIPT="arch/arm/mach-zynq/u-boot.lds"
diff --git a/configs/topic_miamilite_defconfig b/configs/topic_miamilite_defconfig
index 2eba22ed80..e5cfe253f5 100644
--- a/configs/topic_miamilite_defconfig
+++ b/configs/topic_miamilite_defconfig
@@ -16,6 +16,7 @@ CONFIG_DEBUG_UART_CLOCK=100000000
 CONFIG_BOOT_INIT_FILE="board/topic/zynq/zynq-topic-miamilite/ps7_regs.txt"
 CONFIG_DEFAULT_DEVICE_TREE="zynq-topic-miamilite"
 CONFIG_DEBUG_UART=y
+# CONFIG_BOARD_LATE_INIT is not set
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_SYS_CUSTOM_LDSCRIPT=y
 CONFIG_SYS_LDSCRIPT="arch/arm/mach-zynq/u-boot.lds"
diff --git a/include/configs/topic_miami.h b/include/configs/topic_miami.h
index 00c1ca80af..07c020c584 100644
--- a/include/configs/topic_miami.h
+++ b/include/configs/topic_miami.h
@@ -77,7 +77,7 @@
 	"boot_a_script=load ${devtype} ${devnum}:${distro_bootpart} ${scriptaddr} ${prefix}${script}; source ${scriptaddr}\0" \
 	"boot_prefixes=/ /boot/\0" \
 	"boot_scripts=boot.scr\0" \
-	"boot_targets=jtag mmc0 qspi0 usb0 usb1\0" \
+	"boot_targets=mmc0 qspi usb0\0" \
 	"bootcmd_jtag=source $scriptaddr; echo SCRIPT FAILED: continuing...;\0" \
 	"bootcmd_mmc0=setenv devnum 0; run mmc_boot\0" \
 	"bootcmd_mmc1=setenv devnum 1; run mmc_boot\0" \
-- 
2.17.1

